### PR TITLE
Make dashboard test multi-cluster

### PR DESCRIPTION
**Pull request by gargnupur**
Make dashboard test multi-cluster

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[X ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by bianpengyuan — COMMENTED** [review]
Sorry I am not familiar with multi cluster set up. Other than check query in each cluster, is there anything else behind the scene in the test framework that makes this test multi-cluster-ish (e.g. control plane installed in just one cluster? Cross clusters traffic?)

**Comment by gargnupur** [comment]
> Sorry I am not familiar with multi cluster set up. Other than check query in each cluster, is there anything else behind the scene in the test framework that makes this test multi-cluster-ish (e.g. control plane installed in just one cluster? Cross clusters traffic?)

Yes, we have control plane installed on 3 clusters and 2 of the clusters are remote. There is load balancing happening when sending request to any echo server..

File: tests/integration/telemetry/main_test.go [hunk 1]
```diff
@@ -36,7 +36,9 @@ func TestMain(m *testing.M) {
 		Label(label.CustomSetup).
 		Setup(istio.Setup(&i, nil)).
 		Setup(func(ctx resource.Context) (err error) {
-			ingr = i.IngressFor(ctx.Clusters().Default())
+			for _, cl := range ctx.Clusters() {
+				ingr = append(ingr, i.IngressFor(cl))
```
Comment (nmittler) [inline tests/integration/telemetry/main_test.go:40]: Could we just give `istio.Instance` a `Ingresses()` method?

**Review by nmittler — APPROVED** [review]

**Review by stevenctl — APPROVED** [review]

File: tests/integration/telemetry/main_test.go (hunk 1 above)
Comment (gargnupur) [reply tests/integration/telemetry/main_test.go:40]: Yeah that might be good too.. I will refactor some tests in a different pr to try to reduce time.. will handle there?

**Comment by gargnupur** [comment]
Multi-cluster test passed too: https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28741/integ-telemetry-mc-k8s-tests_istio/1326232291348844544

File: tests/integration/telemetry/dashboard_test.go [hunk 2]
```diff
@@ -45,9 +46,10 @@ import (
 
 var (
 	dashboards = []struct {
-		configmap string
-		name      string
-		excluded  []string
+		configmap                       string
+		name                            string
+		excluded                        []string
+		skipValidateOnNonPrimaryCluster bool
```
Comment (douglas-reid) [inline tests/integration/telemetry/dashboard_test.go:52]: nit: i might call this something "simpler" like `primaryClusterOnly` or `requirePrimary`.

The check would then be:

```
if !cl.IsPrimary() && d.primaryClusterOnly {
```

Or:

```
if !cl.IsPrimary() && d.requirePrimary {
```

**Review by douglas-reid — APPROVED** [review]

File: tests/integration/telemetry/dashboard_test.go (hunk 2 above)
Comment (gargnupur) [reply tests/integration/telemetry/dashboard_test.go:52]: Thanks.. FIxed

**Pull request merged**